include/simics/dmllib.h

```diff
@@ -731,7 +731,8 @@ _deserialize_trait_reference(ht_str_table_t *id_info_ht,
     if (unlikely(error != Sim_Set_Ok)) {
         return error;
     }
-    void *trait = ht_lookup_int(vtable_ht, id.id);
+    void *trait = vtable_ht ? ht_lookup_int(vtable_ht, id.id)
+        : (void*)(uintptr_t)0XC0FFEE;
```

Comment:
there's nobody who will drink the coffee? isn't it better to ASSERT, or fail cleanly?

Reply:
No, this is not a case of an error. This is the special case we discussed earlier; empty vtables have no hashtables, and since they're empty the vtable pointer of a trait reference for such a trait is never utilized. Thus, we need some arbitrary _non-`NULL`_ pointer for the vtable pointer during deserialization of trait references with empty trait. Non-`NULL` is important as trait reference serialization checks for a `NULL` vtable pointer to detect a zero-initialized trait reference. We could pick something like `0x1`, but I don't want someone debugging to mistake it for an offset of a null-pointer if they spot it. So I picked `0XC0FFEE`.

Reply:
Right. Can we `#define _DML_EMPTY_VTABLE 0xC0FFEE` with a comment?

Reply:
Sure!